src/librbd/internal.cc

```diff
@@ -1943,6 +1943,19 @@ int mirror_image_disable_internal(ImageCtx *ictx, bool force) {
 	return r;
       }
 
+      if (!old_format) {
+        r = mirror_image_disable_internal(ictx, true);
```

Comment:
We probably want `false` for the force flag.  If the image isn't primary, it is probably being replicated.   If a delete is truly desired, promote it first or delete the primary image and let the deletion propagate over to the peer cluster.